spec/policies/review_policy_spec.rb

```diff
@@ -0,0 +1,27 @@
+require_relative "../../app/policies/review_policy"
+
+RSpec.describe ReviewPolicy, "#authorized_for_review?" do
+  it "returns true if the user has not written a review for the book yet" do
+    reviewer = double(:reviewer)
+    book = double(:book, reviews: [])
+
+    review_authorization =
+      ReviewPolicy.new(book: book, reviewer: reviewer).authorized_for_review?
+
+    expect(review_authorization).to eq true
+  end
+
+  it "returns false if the user has written a review for the book already" do
+    reviewer = double(:reviewer)
+    reviews = double(:reviews)
+    book = double(:book)
+
+    allow(book).to receive(:reviews).and_return reviews
+    allow(reviews).to receive(:map).and_return [reviewer]
```

Comment:
The need to stub `map` here suggests to me that our interface isn't quite right. WDYT?